rootfs/etc/nginx/lua/test/configuration_test.lua

```diff
@@ -0,0 +1,130 @@
+local configuration = require("configuration")
+local unmocked_ngx = _G.ngx
+
+function get_mocked_ngx_env()
+    local _ngx = {
+        status = 100,
+        var = {},
+        req = {
+            read_body = function() end,
+            get_body_data = function() end,
+            get_body_file = function() end,
+        }
+    }
+    setmetatable(_ngx, {__index = _G.ngx})
+    return _ngx
+end
+
+describe("Configuration", function()
+    before_each(function()
+        _G.ngx = get_mocked_ngx_env()
+    end)
+
+    after_each(function()
+        _G.ngx = unmocked_ngx
+        package.loaded["configuration"] = nil
+        configuration = require("configuration")
+    end)
+
+    context("Request method is neither GET nor POST", function()
+        it("should log 'Only POST and GET requests are allowed!'", function()
+            ngx.var.request_method = "PUT"
+            local s = spy.on(ngx, "print")
+            assert.has_no.errors(configuration.call)
+            assert.spy(s).was_called_with("Only POST and GET requests are allowed!")
+        end)
+
+        it("should return a status code of 400'", function()
+            ngx.var.request_method = "PUT"
+            assert.has_no.errors(configuration.call)
+            assert.equal(ngx.status, ngx.HTTP_BAD_REQUEST)
+        end)
+    end)
+
+    context("GET request to /configuration/backends", function()
+        before_each(function()
+            ngx.var.request_method = "GET"
+            ngx.var.request_uri = "/configuration/backends"
+        end)
+
+        it("should call get_backends_data()", function()
+            local s = spy.on(configuration, "get_backends_data")
+            assert.has_no.errors(configuration.call)
+            assert.spy(s).was_called()
+        end)
+
+        it("should call configuration_data:get('backends')", function()
```

Comment:
Why do you care that it specifically calls `configuration_data:get('backends')`? You have access to shared dictionary just add bunch of backends and expect it to be printed that's all, no?

Reply:
That is true, we don't need to mock the shared dictionary anymore 👍